File: my-folder/0121-best-time-to-buy-and-sell-stock/solution.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int ma=0,min=INT_MAX;
        for(int i=0;i<prices.size();i++)
        {
            if(prices[i]<min)
                min=prices[i];
            else
                ma=max(ma,prices[i]-min);
        }
        return ma;
    }
};

File: my-folder/0121-best-time-to-buy-and-sell-stock/solution.java
class Solution {
    public int maxProfit(int[] prices) {

        int cp=prices[0];
        int sp=-1;
        int ci=0;
        int si=-1;
        int profit=0;

        for(int i=1;i<prices.length;i++){
          if(cp>prices[i]){
            cp=prices[i];
          }
          else{
            profit=Math.max(profit,prices[i]-cp);
          }
        }
        
        return profit;
        
    }
}
